README
NAME
    Perl::Critic::logicLAB - a set of Perl::Critic policies implemented and
    used by logicLAB

VERSION
    This documentation describes version 0.05

SYNOPSIS
        % perlcritic --theme logiclab

DESCRIPTION
    Perl::Critic::logicLAB is not a distribution of policies, it is more of
    a meta-package, implemented like the Task distributions.

    This is not the recommended way of doing things, but it does offer the
    following benefits:

    *   Policies can be released separately

    *   Policies can be released without association with
        Perl::Critic::logicLAB until their value and correctness has been
        asserted

    *   You can write a meta-package, listing the policies you want on a
        policy by policy level, using the same schema

POLICIES
    *   Perl::Critic::Policy::logicLAB::ProhibitShellDispatch

    *   Perl::Critic::Policy::logicLAB::ProhibitUseLib

    *   Perl::Critic::Policy::logicLAB::RequireSheBang

    *   Perl::Critic::Policy::logicLAB::RequireVersionFormat

SEE ALSO
    *   <http://logiclab.jira.com/wiki/display/PCLL/Home>

COPYRIGHT & LICENSE
    Copyright 2010-2011 Jonas B. Nielsen (jonasbn), All Rights Reserved.

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.



Changes
Changes file for Perl::Critic::logicLAB

$Id: Changes 8176 2013-08-01 12:55:50Z jonasbn $

Project site changelog:

http://logicLAB.jira.com/browse/PCLL#selectedTab=com.atlassian.jira.plugin.system.project%3Achangelog-panel

0.07 2013-08-01

- Maintenance release, update recommended, since 0.06 distribution was broken

- Fixing build system [PCLL-12]

  https://logiclab.jira.com/browse/PCLL-12


0.06 2013-07-27

- Maintenance release, update not necessary

- Adding a Changes file as part of my Questhub quest adhering to the standard
  described in: CPAN::Changes::Spec [PCLL-8]

  Quest: http://questhub.io/realm/perl/quest/51f0fd8e18ba7dcf4b000040
  Spec: https://metacpan.org/module/CPAN::Changes::Spec
  
  https://logiclab.jira.com/browse/PCLL-8

- Added creation of traditional Makefile.PL to Build.PL [PCLL-9]

  https://logiclab.jira.com/browse/PCLL-9


0.05 2011-05-03

- Update not necessary

- Added new policy: 
  Perl::Critic::Policy::logicLAB::RequireSheBang
  http://logiclab.jira.com/browse/PCLL-7 [PCLL-7]


0.04 2011-04-16

- Update not necessary

- Added new policy: 
  Perl::Critic::Policy::logicLAB::ProhibitShellDispatch
  http://logiclab.jira.com/browse/PCLL-6 [PCLL-6]


0.03 2010-11-24

- Update recommended

- Implemented custom build system
  http://logiclab.jira.com/browse/PCLL-5 [PCLL-5]


0.02 2010-09-14

- Update not necessary
                                
-  Missing tests
   http://logiclab.jira.com/browse/PCLL-4 [PCLL-4]


0.01 2010-09-07

- Wrote documentation
  http://logiclab.jira.com/browse/PCLL-2 [PCLL-2]

- Implementation of Build.PL using the Task scheme
  http://logiclab.jira.com/browse/PCLL-1 [PCLL-1]
  
- Initial release


lib/Perl/Critic/logicLAB.pm
package Perl::Critic::logicLAB;

# $Id: logicLAB.pm 8176 2013-08-01 12:55:50Z jonasbn $

# Courtesy of Adam Kennedy
# http://search.cpan.org/dist/Task/lib/Task.pm

use strict;

use vars qw{$VERSION};

BEGIN {
    $VERSION = '0.07';
}

1;

__END__

=pod

=head1 NAME

Perl::Critic::logicLAB - a set of Perl::Critic policies implemented and used by logicLAB

=head1 VERSION

This documentation describes version 0.05

=head1 SYNOPSIS

    % perlcritic --theme logiclab

=head1 DESCRIPTION

Perl::Critic::logicLAB is not a distribution of policies, it is more of a
meta-package, implemented like the L<Task> distributions.

This is not the recommended way of doing things, but it does offer the following
benefits:

=over

=item * Policies can be released separately

=item * Policies can be released without association with Perl::Critic::logicLAB
until their value and correctness has been asserted

=item * You can write a meta-package, listing the policies you want on a policy
by policy level, using the same schema

=back

=head1 POLICIES

=over

=item * L<Perl::Critic::Policy::logicLAB::ProhibitShellDispatch|Perl::Critic::Policy::logicLAB::ProhibitShellDispatch>

=item * L<Perl::Critic::Policy::logicLAB::ProhibitUseLib|Perl::Critic::Policy::logicLAB::ProhibitUseLib>

=item * L<Perl::Critic::Policy::logicLAB::RequireSheBang|Perl::Critic::Policy::logicLAB::RequireSheBang>

=item * L<Perl::Critic::Policy::logicLAB::RequireVersionFormat|Perl::Critic::Policy::logicLAB::RequireVersionFormat>

=back

=head1 COPYRIGHT & LICENSE

Copyright 2010-2011 Jonas B. Nielsen (jonasbn), All Rights Reserved.

This program is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=cut


Build.PL
# Build.PL for Perl::Critic::logicLAB

# $Id: Build.PL 8176 2013-08-01 12:55:50Z jonasbn $

use strict;
use warnings;
use Module::Build;

our $VERSION = '0.07';

my $build = Module::Build->new(
    dist_author   => 'Jonas B. Nielsen (jonasbn), <jonasbn@cpan.org>',
    module_name   => 'Perl::Critic::logicLAB',
    license       => 'artistic',
    create_makefile_pl => 'traditional',
    create_readme => 1,
    requires      => {
        'Perl::Critic::Policy::logicLAB::ProhibitUseLib'        => '0',
        'Perl::Critic::Policy::logicLAB::RequireVersionFormat'  => '0',
        'Perl::Critic::Policy::logicLAB::ProhibitShellDispatch' => '0',
        'Perl::Critic::Policy::logicLAB::RequireSheBang'        => '0',
    },
    meta_merge => {
        resources => {
            homepage => 'http://logiclab.jira.com/wiki/display/PCLL/Home',
            bugtracker =>
                'http://rt.cpan.org/NoAuth/Bugs.html?Dist=Perl-Critic-logicLAB',
            repository => 'http://logicLAB.jira.com/svn/PCLL',
            ProjectChangelog =>
                'http://logiclab.jira.com/browse/PCLL#selectedTab=com.atlassian.jira.plugin.system.project%3Achangelog-panel',
            ProjectSite => 'http://logicLAB.jira.com/browse/PCLL',
            ProjectWiki => 'http://logiclab.jira.com/wiki/display/PCLL/Home',
        }
    },
);

$build->create_build_script();
